package.json

```diff
@@ -20,77 +12,146 @@
     "build": "ember build --environment=production",
-    "lint": "npm-run-all --aggregate-output --continue-on-error --parallel \"lint:!(fix)\"",
-    "lint:fix": "npm-run-all --aggregate-output --continue-on-error --parallel lint:*:fix",
+    "lint": "concurrently \"npm:lint:*(!fix)\" --names \"lint:\"",
+    "lint:css": "stylelint \"**/*.css\"",
+    "lint:css:fix": "concurrently \"npm:lint:css -- --fix\"",
+    "lint:fix": "concurrently \"npm:lint:*:fix\" --names \"fix:\"",
     "lint:hbs": "ember-template-lint .",
     "lint:hbs:fix": "ember-template-lint . --fix",
     "lint:js": "eslint . --cache",
     "lint:js:fix": "eslint . --fix",
+    "lint:types": "tsc --noEmit",
+    "prepack": "tsc --project tsconfig.declarations.json",
+    "postpack": "rimraf declarations",
     "start": "ember serve",
-    "test": "npm-run-all lint test:*",
+    "test": "concurrently \"npm:lint\" \"npm:test:*\" --names \"lint,test:\"",
     "test:ember": "ember test",
     "test:ember-compatibility": "ember try:each"
   },
   "dependencies": {
-    "ember-cli-babel": "^7.26.6",
-    "ember-cli-htmlbars": "^6.0.0",
+    "@babel/core": "^7.23.2",
+    "ember-auto-import": "^2.6.3",
+    "ember-cli-babel": "^8.2.0",
+    "ember-cli-htmlbars": "^6.3.0",
     "ember-feature-flags": "^6.0.0",
-    "ember-local-storage": "^2.0.1",
-    "ember-truth-helpers": "^3.0.0"
+    "ember-local-storage": "^2.0.6",
+    "ember-truth-helpers": "^4.0.3"
   },
   "devDependencies": {
-    "@babel/core": "^7.18.10",
-    "@babel/eslint-parser": "^7.18.9",
-    "@babel/plugin-proposal-decorators": "^7.18.10",
     "@ember/optional-features": "^2.0.0",
-    "@ember/test-helpers": "^2.4.2",
-    "@embroider/test-setup": "^1.6.0",
-    "@glimmer/component": "^1.0.4",
-    "@glimmer/tracking": "^1.0.4",
+    "@ember/string": "^3.1.1",
+    "@ember/test-helpers": "^3.2.0",
+    "@embroider/test-setup": "^3.0.2",
+    "@glimmer/component": "^1.1.2",
+    "@glimmer/tracking": "^1.1.2",
+    "@glint/environment-ember-loose": "^1.2.1",
+    "@glint/template": "^1.2.1",
+    "@tsconfig/ember": "^3.0.2",
+    "@types/ember": "^4.0.8",
+    "@types/ember__application": "^4.0.9",
+    "@types/ember__array": "^4.0.7",
+    "@types/ember__component": "^4.0.19",
+    "@types/ember__controller": "^4.0.9",
+    "@types/ember__debug": "^4.0.6",
+    "@types/ember__destroyable": "^4.0.3",
+    "@types/ember__engine": "^4.0.8",
+    "@types/ember__error": "^4.0.4",
+    "@types/ember__helper": "^4.0.4",
+    "@types/ember__modifier": "^4.0.7",
+    "@types/ember__object": "^4.0.9",
+    "@types/ember__owner": "^4.0.7",
+    "@types/ember__polyfills": "^4.0.4",
+    "@types/ember__routing": "^4.0.17",
+    "@types/ember__runloop": "^4.0.7",
+    "@types/ember__service": "^4.0.6",
+    "@types/ember__string": "^3.16.3",
+    "@types/ember__template": "^4.0.4",
+    "@types/ember__test": "^4.0.4",
+    "@types/ember__utils": "^4.0.5",
+    "@types/qunit": "^2.19.7",
+    "@types/rsvp": "^4.0.6",
```

Comment:
For my personal knowledge, is it automatically generated? How you handle updates (like if you update `qunit` package to a new version, you run a specific command to updates `types` too or it need to be done manually?

Reply:
It is from the blueprint:
https://github.com/ember-cli/ember-cli/blob/v5.4.0/blueprints/app/files/package.json

which then gets transformed by the add-on blueprint:
https://github.com/ember-cli/ember-cli/blob/v5.4.0/blueprints/addon/index.js